rhino/src/main/java/org/mozilla/javascript/NativeString.java

```diff
@@ -772,6 +780,56 @@ else if (Normalizer.Form.NFC.name().equals(formStr))
 
                         return str.substring(k, k + 1);
                     }
+                case Id_isWellFormed:
+                    {
+                        String str = ScriptRuntime.toString(requireObjectCoercible(cx, thisObj, f));
```

Comment:
This is minor but it will likely help with performance in a bunch of cases -- how about using "toCharSequence" here instead of "toString"? That would mean that concatenated strings here don't need to be re-concatenated, and would work fine since it looks like you're just iterating character-by-character anyway.